lib/android/app/src/main/java/com/reactnativenavigation/anim/TopBarAnimator.java

```diff
@@ -21,15 +23,23 @@
     private final AccelerateInterpolator accelerateInterpolator = new AccelerateInterpolator();
 
     private TopBar topBar;
+    private String stackId;
     private AnimatorSet hideAnimator;
     private AnimatorSet showAnimator;
 
     public TopBarAnimator(TopBar topBar) {
         this.topBar = topBar;
     }
 
+    public TopBarAnimator(TopBar topBar, @Nullable String stackId) {
+        this.topBar = topBar;
+        this.stackId = stackId;
+    }
+
     public void show(AnimationOptions options) {
-        if (options.hasValue()) {
+        if (options.hasValue() && options.id.hasValue() && options.id.get().equals(stackId)) {
+            showAnimator = options.getAnimation(topBar);
```

Comment:
Line 41 equals line 43... is this conditional needed?

Reply:
No. The difference is, that first condition is true, if you have provided id and it's the same. And second condition true if you didn't provide id at all